apps/ensadmin/src/components/recent-registrations/components.tsx

```diff
@@ -143,10 +150,52 @@ interface RegistrationRowProps {
   namespaceId: ENSNamespaceId;
 }
 
+/**
+ * NOTE: This function is a "dirty-hack".
+ *
+ * TODO: Update our indexed data model for registrations so that this dirty hack
+ *       is no longer needed.
+ */
+const guessChainIdFromRegisteredName = (name: Name, namespaceId: ENSNamespaceId): ChainId => {
+  const labels = name.split(".");
+
+  if (labels.length === 3) {
+    if (name.endsWith(".base.eth")) {
+      // name is a direct subname of .base.eth
+      // we will therefore assume it occured within Basenames.
+      // NOTE: this assumption is not necessarily true, nothing technically stops
+      // subnames of base.eth from occurring on the ENS root chain. And some have.
+      // Therefore, this is only a "dirty-hack" approximation.
+      const basenames = maybeGetDatasource(namespaceId, DatasourceNames.Basenames);
+
+      if (basenames) return basenames.chain.id;
+
+      // basenames is undefined for the namespace, therefore fallback
+      // to the ENS root chain
+    } else if (name.endsWith(".linea.eth")) {
+      // name is a direct subname of .linea.eth
+      // we will therefore assume it occured within Lineanames.
+      // NOTE: this assumption is not necessarily true, nothing technically stops
+      // subnames of linea.eth from occurring on the ENS root chain. And some have.
+      // Therefore, this is only a "dirty-hack" approximation.
+      const linea = maybeGetDatasource(namespaceId, DatasourceNames.Lineanames);
+
+      if (linea) return linea.chain.id;
+
+      // basenames is undefined for the namespace, therefore fallback
+      // to the ENS root chain
+    }
+  }
+
+  // fallback to assuming the the registration occurred on the ENS root chain
+  return getENSRootChainId(namespaceId);
```

Comment:
I would make this function

```ts
function guessChainIdFromRegisteredName(name: Name): ChainId | null {
   // ...
}
```

and implement the defaulting within the component like

```ts
const chainId = guessChainIdFromRegisteredName(registration.name) ?? getENSRootChainId(namespaceId)
```

so that the 'guess' verbiage reads better, and we don't need to pass the namespace to the helper

also that helper could/should live in an ensadmin/lib file somewhere, perhaps `guess-registration-chain-id.ts`